LibreNMS/Modules/Slas.php

```diff
@@ -0,0 +1,614 @@
+<?php
+/**
+ * SLA.php
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * @link       https://www.librenms.org
+ */
+
+namespace LibreNMS\Modules;
+
+use App\Models\Sla;
+use LibreNMS\Interfaces\Module;
+use LibreNMS\OS;
+use LibreNMS\RRD\RrdDefinition;
+use LibreNMS\Util\IP;
+
+class Slas implements Module
+{
+    //use SyncsModels;
+
+    /**
+     * Discover this module. Heavier processes can be run here
+     * Run infrequently (default 4 times a day)
+     *
+     * @param OS $os
+     */
+    public function discover(OS $os)
+    {
+        $device = $os->getDeviceArray();
+
+        if ($device['os_group'] == 'cisco') {
+            $this->discoverSlas($device);
+        } elseif ($device['os'] == 'junos') {
+            $this->discoverRpms($device);
+        }
+        // } else if ($device['os'] == 'huawei')
```

Comment:
That's for me, I guess 👍

Reply:
Yes you said you need the feature for Huawei. It is probably not the right way to target Huawei's device though. I let you implement this part.